src/programs/console_test/console_test.cpp

```diff
@@ -1789,6 +1791,72 @@ void MyTestApp::TestRunProfileDiskOperations( ) {
     EndTest( );
 }
 
+void MyTestApp::TestCTFNodes( ) {
+    BeginTest("CTF Nodes");
+
+    Curve ctf_curve1;
+    Curve ctf_curve2;
+    ctf_curve1.SetupXAxis(0.0, 0.5, 500);
+    ctf_curve2.SetupXAxis(0.0, 0.5, 500);
+
+    CTF ctf1(300, 2.7, 0.07, 5000, 5000, 0, 1.0, 0.0);
+
+    ctf_curve1.SetYToConstant(1.0);
+    ctf_curve2.SetYToConstant(1.0);
+    ctf_curve1.ApplyCTF(ctf1);
+    // Apply wwith Thicknes returns power of two
+    ctf_curve1.MultiplyBy(ctf_curve1);
+    ctf_curve2.ApplyCTFWithThickness(ctf1);
+
+    if ( ctf_curve1.YIsAlmostEqual(ctf_curve2) == false ) {
+        FailTest;
+    }
+
+    CTF ctf2;
+    // CTF with a sample thickness parameter of 100.0
+    ctf2.Init(300, 2.7, 0.07, 5000, 5000, 0, 1.0, 0.0, 100.0);
+
+    ctf_curve1.SetYToConstant(1.0);
+    ctf_curve2.SetYToConstant(1.0);
+    ctf_curve1.ApplyCTF(ctf2);
+    // Apply wwith Thicknes returns power of two
+    ctf_curve1.MultiplyBy(ctf_curve1);
+
+    ctf_curve2.ApplyCTFWithThickness(ctf2);
+
+    // CTF is different when thickness is 100
+    if ( ctf_curve1.YIsAlmostEqual(ctf_curve2) == true ) {
+        FailTest;
+    }
+
+    // Test manually integrating ctf and compare with thickness formula
+    ctf_curve1.SetYToConstant(0.0);
+    Curve ctf_curve3;
+    ctf_curve3.SetupXAxis(0.0, 0.5, 500);
+    int counter = 0;
+
+    for ( float z_level = -495.0; z_level < 500.0; z_level = z_level + 10.0f ) {
+        ctf1.Init(300, 2.7, 0.07, 5000 + z_level, 5000 + z_level, 0, 1.0, 0.0, 0.0);
+        counter++;
+        ctf_curve3.SetYToConstant(1.0);
+        ctf_curve3.ApplyCTF(ctf1);
+        ctf_curve3.MultiplyBy(ctf_curve3);
+        ctf_curve1.AddWith(&ctf_curve3);
+    }
+
+    // Now want to compare ctf_curve1 with ctf_curve2, but FloatIsAlmostEqual is to stringent.
+    // The code below is a hack to get around this. Ideally, FloatIsAlmostEqual should be modified to allow custom tolerances.
```

Comment:
I added this a while back in functions.h - it may not be the best way, and it relative, not absolute error. But anyway ...

```cpp
template <typename T>
inline bool RelativeErrorIsLessThanEpsilon(T reference, T test_value, T epsilon = 0.0001) {
    return (std::abs((reference - test_value) / reference) < epsilon);
}
```

It looks like in my repo, it has changed a bit

```cpp
template <typename T>
bool RelativeErrorIsLessThanEpsilon(T reference, T test_value, bool print_if_failed = true, T epsilon = 0.0001) {

    bool ret_val;
    // I'm not sure if this is the best way to guard against very small division
    if ( abs(reference) < epsilon || abs(test_value) < epsilon )
        ret_val = (std::abs((reference - test_value)) < epsilon);
    else
        ret_val = (std::abs((reference - test_value) / reference) < epsilon);

    if ( print_if_failed && ! ret_val ) {
        std::cerr << "RelativeErrorIsLessThanEpsilon failed: " << reference << " " << test_value << " " << epsilon << " " << std::abs((reference - test_value) / reference) << std::endl;
    }
    return ret_val;
};
```